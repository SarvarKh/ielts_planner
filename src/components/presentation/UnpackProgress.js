import Navbar from '../presentation/Navbar';
import Footer from '../presentation/Footer';
import MyChart from '../container/MyChart';
import { CircularProgressbar } from 'react-circular-progressbar';

const UnpackProgress = ({results}) => {
    let current_scores = results[results.length-1];
    return (
    <div>
        <Navbar title = "Progress" />
            <main>
                <h2>Progress page</h2>
                <MyChart results={results} />
                {
                    current_scores
                    ? <>
                        <h3>Current scores</h3>
                        <div className="prog-circle-score">
                            <CircularProgressbar value={current_scores.overall_score} maxValue={9} text={`Overall Score\n ${current_scores.overall_score}`}/>
                        </div>
                        <div>Speaking: {current_scores.speaking_score}</div>
                        <div>Listening: {current_scores.listening_score}</div>
                        <div>Writing: {current_scores.writing_score}</div>
                        <div>Reading: {current_scores.reading_score}</div>
                    </>
                    : <div>Loading...</div>
                }
            </main>
        <Footer />
    </div>
    );
}

export default UnpackProgress;