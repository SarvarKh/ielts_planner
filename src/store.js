import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from './reducers/rootReducer';

const initialState = {
//   detailMeal: {
//     detailMeal: {},
//   },
    results: [],
    current_user: {}
};

const middleware = [thunk];

const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(
    applyMiddleware(...middleware),
  ),
);

export default store;
