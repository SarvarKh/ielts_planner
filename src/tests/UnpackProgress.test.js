import React from 'react';
import { Provider } from 'react-redux';
import renderer, { act } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../store';
import UnpackProgress from '../components/presentation/UnpackProgress';

it('renders correctly', () => {
  const results = [];
  let tree;
  act(() => {
    tree = renderer.create(
      <Provider store={store}>
        <React.StrictMode>
          <BrowserRouter>
            <UnpackProgress results={results} />
          </BrowserRouter>
        </React.StrictMode>
      </Provider>,
    ).toJSON();
  });

  expect(tree).toMatchSnapshot();
});
