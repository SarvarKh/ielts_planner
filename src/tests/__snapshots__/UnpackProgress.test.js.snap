// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav>
    <div
      className="n-title"
    >
      Progress
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-power-off"
      />
    </button>
  </nav>
  <main>
    <h3
      className="g-title"
    >
      Analyze your progress against plan.
    </h3>
    <h2
      className="t-empty"
    >
      Loading...
    </h2>
    <div
      className="g-description"
    >
      <div>
        Blue - actual data
      </div>
      <div>
        Red - planned data
      </div>
    </div>
    <div
      className="t-empty"
    >
      Please add scores to see the charts.
    </div>
  </main>
  <footer>
    <a
      aria-current={null}
      href="/measurement"
      onClick={[Function]}
    >
      <span
        className="f-icons"
      >
        <svg
          className="h-10 w-10"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 3a2 2 0 00-2 2v2a2 2 0 002 2h2a2 2 0 002-2V5a2 2 0 00-2-2H5zM5 11a2 2 0 00-2 2v2a2 2 0 002 2h2a2 2 0 002-2v-2a2 2 0 00-2-2H5zM11 5a2 2 0 012-2h2a2 2 0 012 2v2a2 2 0 01-2 2h-2a2 2 0 01-2-2V5zM14 11a1 1 0 011 1v1h1a1 1 0 110 2h-1v1a1 1 0 11-2 0v-1h-1a1 1 0 110-2h1v-1a1 1 0 011-1z"
          />
        </svg>
      </span>
      <div
        className="f-text"
      >
        Add measure
      </div>
    </a>
    <a
      aria-current={null}
      href="/track"
      onClick={[Function]}
    >
      <span
        className="f-icons"
      >
        <svg
          className="h-10 w-10"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M12 7a1 1 0 110-2h5a1 1 0 011 1v5a1 1 0 11-2 0V8.414l-4.293 4.293a1 1 0 01-1.414 0L8 10.414l-4.293 4.293a1 1 0 01-1.414-1.414l5-5a1 1 0 011.414 0L11 10.586 14.586 7H12z"
            fillRule="evenodd"
          />
        </svg>
      </span>
      <div
        className="f-text"
      >
        Track it
      </div>
    </a>
    <a
      aria-current={null}
      href="/progress"
      onClick={[Function]}
    >
      <span
        className="f-icons"
      >
        <svg
          className="h-10 w-10"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 10a8 8 0 018-8v8h8a8 8 0 11-16 0z"
          />
          <path
            d="M12 2.252A8.014 8.014 0 0117.748 8H12V2.252z"
          />
        </svg>
      </span>
      <div
        className="f-text"
      >
        Your progress
      </div>
    </a>
    <a
      aria-current={null}
      href="/more"
      onClick={[Function]}
    >
      <span
        className="f-icons"
      >
        <svg
          className="h-10 w-10"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 10a2 2 0 11-4 0 2 2 0 014 0zM12 10a2 2 0 11-4 0 2 2 0 014 0zM16 12a2 2 0 100-4 2 2 0 000 4z"
          />
        </svg>
      </span>
      <div
        className="f-text"
      >
        More
      </div>
    </a>
  </footer>
</div>
`;
